lib/ClangImporter/ImportEnumInfo.cpp

```diff
@@ -259,10 +259,19 @@ ImportedType importer::findOptionSetEnum(clang::QualType type,
   if (!clangEnum)
     return ImportedType();
 
-  // If this fails, it means that we need a stronger predicate for
+  // Assert that the typedef has the same underlying integer representation as
+  // the enum we think it assigns a type name to.
+  //
+  // If these fails, it means that we need a stronger predicate for
   // determining the relationship between an enum and typedef.
-  assert(clangEnum.value()->getIntegerType()->getCanonicalTypeInternal() ==
-         typedefType->getCanonicalTypeInternal());
+  if (auto *tdEnum =
+          dyn_cast<clang::EnumType>(typedefType->getCanonicalTypeInternal())) {
+    ASSERT(clangEnum.value()->getIntegerType()->getCanonicalTypeInternal() ==
```

Comment:
I guess having these as `ASSERT` over `assert` might cause these to start failing in release builds, which might be a bit risky for 6.2... I don't have a strong opinion on this.

Reply:
I can make these `assert()` and then change them to `ASSERT()` in a `main`-only follow-up, to de-risk.

Reply:
Actually, I'm no longer using `assert()` since those are prone to "unused variable" warnings. I'm now using `ASSERT()`, guarded by `#ifndef NDEBUG`, which preserves the behavior of `assert()`; the latest commit fa5c4f2 removes the `#ifndef NDEBUG`, but I do not plan on cherry-picking that to 6.2.

Reply:
I see, thanks for explaining!